### PR TITLE
feat[WIP]: Filters

### PR DESCRIPTION
**Pull request by saadahmadhere**
This PR mainly focuses on:
1. Sort by price filter.
2. Filter by category.
3. Filter by rating.
4. Filter by price.
5. Filter by stock availability.

### CODE REVIEW

File: src/Components/Filters/Filters.css [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+.price,
+.rating,
+.sort,
+.availability {
+  display: none;
+}
+
+/* products */
+.product_title {
+  margin: var(--size-4) var(--size-1) var(--size-8) var(--size-4);
```
Comment (hinanshisuthar) [inline src/Components/Filters/Filters.css:43]: Good use of predefined variables wherever possible

File: src/Components/Filters/Filters.jsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+      <div className="category">
+        <h4 className="h4 gray_title mb_4">Categories</h4>
+        <div className="category_item">
+          <input
+            type="checkbox"
+            name="category"
+            id="earphones"
+            onChange={() =>
+              filterDispatch({ type: "CATEGORY", payload: "ear phones" })
```
Comment (hinanshisuthar) [inline src/Components/Filters/Filters.jsx:65]: It is a good practice if you keep you payload name in CAPS with underscore in between,

File: src/Components/Filters/RatingFilter.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    filterDispatch,
+  } = useFilter();
+
+  return [...Array(5)].map((_, i) => (
+    <span
+      className="mr_1"
+      style={{ cursor: "pointer", color: "#ff9529" }}
+      onClick={() => {
+        filterDispatch({ type: "FILTER_BY_RATING", payload: i + 1 });
```
Comment (hinanshisuthar) [inline src/Components/Filters/RatingFilter.jsx]: If you want to pass index, use the word index. It is not clear what you are trying to pass as a payload

**Review by hinanshisuthar — COMMENTED** [review]
- Suggestions given to improve code quality
- Overall good use of variables
- Liked how simple/neat logic you preferred instead of deep ones
- Good to merge from my side

**Review by abhij1607 — COMMENTED** [review]
Very neat & clean code | | Good practices followed | | Modular files

File: src/Components/Filters/RatingFilter.jsx (hunk 3 above)
Comment (saadahmadhere) [reply src/Components/Filters/RatingFilter.jsx]: fixed in [fix variable naming](https://github.com/saadahmadhere/elmart/pull/8/commits/742e340e0a4a6271e8bea46ae3c818324ec909fc)

File: src/Components/Filters/Filters.jsx (hunk 2 above)
Comment (saadahmadhere) [reply src/Components/Filters/Filters.jsx:65]: will take care of it.

**Pull request merged**